xarray/backends/netCDF4_.py

```diff
@@ -402,8 +436,13 @@ def open(
                 lock = combine_locks([base_lock, get_write_lock(filename)])
 
         kwargs = dict(
-            clobber=clobber, diskless=diskless, persist=persist, format=format
+            clobber=clobber,
+            diskless=diskless,
+            persist=persist,
+            format=format,
         )
+        if auto_complex is not None:
+            kwargs["auto_complex"] = auto_complex
```

Comment:
Is this to avoid passing `auto_complex` to the h5netcdf engine?

Reply:
To not feed it to old versions which can't digest it.